src/Stratis.Bitcoin.Features.Consensus/ConsensusStats.cs

```diff
@@ -107,13 +108,18 @@ public void BenchStats()
             this.logger.LogInformation(benchLogs.ToString());
         }
 
-        protected override void OnNextCore(Block value)
+        protected override void OnNextCore(ChainedHeaderBlock chainedHeaderBlock)
         {
-            if (this.dateTimeProvider.GetUtcNow() - this.lastSnapshot.Taken > TimeSpan.FromSeconds(5.0))
-            {
-                if (this.initialBlockDownloadState.IsInitialBlockDownload())
-                    this.BenchStats();
-            }
+            if (this.LastSnapshotWasWithinLast5Seconds()) 
+                return;
+            
+            if (this.initialBlockDownloadState.IsInitialBlockDownload())
+                this.BenchStats();
+        }
+
+        private bool LastSnapshotWasWithinLast5Seconds()
```

Comment:
Please move this logic back to if because it's outside of the scope of the PR and ConsensusStats will be completely rewritten after activation is merged

Reply:
In that case, there is no harm in it staying, but I see your point and will revert anyway.